src/spaces/graph.jl

```diff
@@ -127,3 +138,23 @@ function nearby_positions(
     filter!(i -> i != position, output)
 end
 
+#######################################################################################
+# Mutable graph functions
+#######################################################################################
+"""
+    rem_node!(n::Int, model::ABM{A, <: GraphSpace})
+Remove node (i.e. position) `n` from the model's graph. All agents in that node are killed.
+
+**Warning:** LightGraphs.jl (and a result Agents.jl) swaps the index of the last node with
+that of the one to be removed, while every other node remains as is. This means that
+when doing `rem_node!(n, model)` the last node becomes the `n`-th node while the previous
+`n` node (and all its edges and agents) are deleted.
+"""
+function rem_node!(n::Int, model::ABM{<:AbstractAgent, <: GraphSpace})
+    for a ∈ agents_in_position(n, model); kill_agent!(a, model); end
+    s = model.space.s
+    V = nv(model)
+    s[V], s[n] = s[n], s[V]
+    pop!(s)
+    LightGraphs.rem_vertex!(model.space.graph, n)
+end
```

Comment:
@yanyuan-wang does this seem correct to you?

Reply:
Yes, that looks good!